IntegrationTest/AuthenticationEndpoints.cs

```diff
@@ -0,0 +1,160 @@
+using API;
+using Core.Shared;
+using FluentResults;
+using Infrastructure.Authentication.Contracts;
+using Infrastructure.Authentication.Models;
+using IntegrationTest.Setup;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using NSubstitute;
+
+namespace IntegrationTest;
+
+[Collection(CollectionDefinitions.NonParallelCollectionName)]
+public class AuthenticationEndpoints : IClassFixture<TestWebApplicationFactory<Program>>
+{
+	private readonly HttpClient _client;
+	private readonly TestWebApplicationFactory<Program> _factory;
+	public AuthenticationEndpoints(TestWebApplicationFactory<Program> factory)
+	{
+		_factory = factory;
+		_client = factory.CreateClient();
+	}
+	// Refresh token
+	[Fact]
+	public async Task RefreshToken_ShouldReturn_BadRequest_RefreshTokenNotValid()
+	{
+		using var scope = _factory.Services.CreateScope();
+		var tokenRepo = scope.ServiceProvider.GetService<ITokenRepository>();
+		tokenRepo!.GetAccessTokenByRefreshTokenAsync(Arg.Any<string>()).Returns(Result.Fail("failed"));
+		var dto = new RefreshDto("token");
+
+		var response = await _client.PostAsJsonAsync("/refresh", dto);
+		
+		
+		Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);	
+	}
+	
+	[Fact]
+	public async Task RefreshToken_ShouldReturn_BadRequest_UserDoesntExist()
+	{
+		using var scope = _factory.Services.CreateScope();
+		var tokenRepo = scope.ServiceProvider.GetService<ITokenRepository>();
+		var userRepo = scope.ServiceProvider.GetService<IUserRepository>();
+		var rf = new RefreshToken{UserId = 1};
+		tokenRepo!.GetAccessTokenByRefreshTokenAsync(Arg.Any<string>()).Returns(Result.Ok(rf));
+		userRepo!.GetRolesByUserIdAsync(Arg.Any<int>()).Returns(new List<Role>());
+		var dto = new RefreshDto("token");
+
+		var response = await _client.PostAsJsonAsync("/refresh", dto);
+		
+		
+		Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);	
+	}
+	[Fact]
+	public async Task RefreshToken_ShouldReturn_OkLoginResponse()
+	{
+		using var scope = _factory.Services.CreateScope();
+		var tokenRepo = scope.ServiceProvider.GetService<ITokenRepository>();
+		var userRepo = scope.ServiceProvider.GetService<IUserRepository>();
+		var rf = new RefreshToken{UserId = 1};
+		tokenRepo!.GetAccessTokenByRefreshTokenAsync(Arg.Any<string>()).Returns(Result.Ok(rf));
+		userRepo!.GetRolesByUserIdAsync(Arg.Any<int>()).Returns(new List<Role>{new Role{RoleName = "Instructor"}});
+		var dto = new RefreshDto("token");
+
+		var response = await _client.PostAsJsonAsync("/refresh", dto);
+		
+		
+		Assert.True(response.IsSuccessStatusCode);
+	}
+	
+	// login
+	[Fact]
+	public async Task Login_ShouldReturn_WrongPassword()
+	{
+		using var scope = _factory.Services.CreateScope();
+		var userRepo = scope.ServiceProvider.GetService<IUserRepository>();
+		var x = new PasswordHasher<User>();
+		var user = new User { Id = 1, PasswordHash = "password", Email = "email@email.com", Name = "name", CreatedAt = DateTime.UtcNow};
```

Comment:
does `PasswordHash` act as a salt for hashing?

I notice it does not align with the password values given later in the testcase